### PR TITLE
NF access to NDA

### PR DESCRIPTION
**Pull request by yarikoptic**
Requires piping of user/password into a helper which requests AWS S3 credentials from the NDA

It is untested still, initial attempts to access some found on the web s3 link failed with permission denied, which might be just a legit access denied

edit 1: this PR absorbing few others now

### CODE REVIEW

File: datalad/support/third/nda_aws_token_generator.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+## NDA AWS Token Generator
+## Author: NIMH Data Archives
+##         http://ndar.nih.gov
+## License: MIT
```
Comment (yarikoptic) [inline datalad/support/third/nda_aws_token_generator.py]: was wondering:  https://github.com/NDAR/nda_aws_token_generator/issues/2

**Comment by yarikoptic** [comment]
I wish we have had merged it as is ;)  but we better don't at least not until release/OHBM demos I guess.

**Comment by yarikoptic** [comment]
ok @datalad/developers  -- I think this one is ready for review/merge before it becomes too stale again.  Missing coverage is primarily for functionality requiring S3 access  (we are yet to establish it within travis) or access to NDA.  I could theoretically mock it up somehow but since not sure if we would ever push/finalize support of NDA, I am not keen on sweating on it atm, thus please review/merge

**Comment by hanke** [comment]
I am still largely unfamiliar with the crawler codebase, but LGTM...

**Comment by yarikoptic** [comment]
I would wholeheartedly welcome your familiarization ;)  so many things could be improved

**Comment by yarikoptic** [comment]
since no objections -- merging

**Pull request merged**